### PR TITLE
Try fix problem in base58_encode

### PR DESCRIPTION
**Pull request by pavel-kokolemin**

### CODE REVIEW

**Comment by sergey-chernikov** [comment]
Please submit this patch to upstream

File: cppForSwig/BtcUtils.h [hunk 1]
```diff
@@ -1760,11 +1760,11 @@ class BtcUtils
    static std::string base58_encode(const BinaryData& payload)
    {
       size_t size = payload.getSize() * 2;
-      size_t return_size;
       std::string b58_str;
       b58_str.resize(size);
+      size_t return_size = b58_str.size();
       if (!btc_base58_encode(
-         (char*)b58_str.c_str(), &return_size,
+         &b58_str[0], &return_size,
```
Comment (Ation) [inline cppForSwig/BtcUtils.h]: Please use c_str. in terms of standard - it should be exactly the same, but c_str is more "natural".

Comment (Ation) [reply cppForSwig/BtcUtils.h]: Maybe we should submit this to an upstream?

Comment (pavel-kokolemin) [reply cppForSwig/BtcUtils.h]: > Please use c_str. in terms of standard - it should be exactly the same, but c_str is more "natural".

Why? I think we already discussed this: https://en.cppreference.com/w/cpp/string/basic_string/c_str
> Writing to the character array accessed through c_str() is undefined behavior.

Comment (sergey-chernikov) [reply cppForSwig/BtcUtils.h]: BTW, why at all do we need to pass data from `std::string` to `btc_base58_encode`? If it accepts a C string, why not to allocate and pass a `char[]` there and then construct return value from it after the call?

Comment (pavel-kokolemin) [reply cppForSwig/BtcUtils.h]: > BTW, why at all do we need to pass data from `std::string` to `btc_base58_encode`?

Using `std::string` is faster (no need for extra allocation and copying).

**Pull request merged**